tests/src/kakarot/test_kakarot.py

```diff
@@ -405,6 +408,182 @@ def test_failing_contract(self, cairo_run):
+            ):
+                cairo_run(
+                    "test__eth_send_raw_transaction",
+                    tx_data_len=len(tx_data),
+                    tx_data=tx_data,
+                )
+
+        @SyscallHandler.patch("Kakarot_block_gas_limit", 0)
+        @given(maxFeePerGas=integers(min_value=2**128, max_value=DEFAULT_PRIME - 1))
+        def test_raise_max_fee_per_gas_too_high(self, cairo_run, maxFeePerGas):
+            tx = {
+                "type": 2,
+                "gas": 100_000,
+                "maxFeePerGas": maxFeePerGas,
+                "maxPriorityFeePerGas": 3_000_000_000,
+                "data": "0x616263646566",
+                "nonce": 34,
+                "to": "0x09616C3d61b3331fc4109a9E41a8BDB7d9776609",
+                "value": 0x5AF3107A4000,
+                "accessList": [],
+                "chainId": CHAIN_ID,
+            }
+            tx_data = list(rlp_encode_signed_data(tx))
+
+            with (
+                SyscallHandler.patch("IAccount.get_nonce", lambda _, __: [tx["nonce"]]),
+                cairo_error(message="Max fee per gas too high"),
+            ):
+                cairo_run(
+                    "test__eth_send_raw_transaction",
+                    tx_data_len=len(tx_data),
+                    tx_data=tx_data,
+                )
+
+        @SyscallHandler.patch("Kakarot_block_gas_limit", 0)
+        @pytest.mark.parametrize("tx", TRANSACTIONS)
+        def test_raise_transaction_gas_limit_too_high(self, cairo_run, tx):
+            tx_data = list(rlp_encode_signed_data(tx))
+
+            with (
+                SyscallHandler.patch("IAccount.get_nonce", lambda _, __: [tx["nonce"]]),
+                cairo_error(message="Transaction gas_limit > Block gas_limit"),
+            ):
+                cairo_run(
+                    "test__eth_send_raw_transaction",
+                    tx_data_len=len(tx_data),
+                    tx_data=tx_data,
+                )
+
+        @SyscallHandler.patch("Kakarot_block_gas_limit", TRANSACTION_GAS_LIMIT)
+        @SyscallHandler.patch("Kakarot_base_fee", TRANSACTION_GAS_LIMIT * 10**10)
+        @pytest.mark.parametrize("tx", TRANSACTIONS)
+        def test_raise_max_fee_per_gas_too_low(self, cairo_run, tx):
+            tx_data = list(rlp_encode_signed_data(tx))
+
+            with (
+                SyscallHandler.patch("IAccount.get_nonce", lambda _, __: [tx["nonce"]]),
+                cairo_error(message="Max fee per gas too low"),
+            ):
+                cairo_run(
+                    "test__eth_send_raw_transaction",
+                    tx_data_len=len(tx_data),
+                    tx_data=tx_data,
+                )
+
+        @composite
+        def max_priority_fee_too_high(draw):
+            maxFeePerGas = draw(integers(min_value=0, max_value=2**128 - 1))
+            maxPriorityFeePerGas = draw(integers(min_value=0, max_value=2**128 - 1))
+            assume(maxFeePerGas < maxPriorityFeePerGas)
```

Comment:
why not using the max_fee_per_gas as a min in the drawing of max_priority_fee_per_gas ?